lib/proxy/pluggable.rb

```diff
@@ -1,131 +1,137 @@
 module Proxy::Pluggable
-  def self.included(base)
-    base.send(:include, InstanceMethods)
-    base.send(:extend, ClassMethods)
+  attr_reader :plugin_name, :version, :after_activation_blk, :class_loader
+
+  # Methods below define DSL for defining plugins
+  def after_activation(&blk)
```

Comment:
Not at the moment, although it can be deprecated if I add support for blocks as means of loading classes, specifying di wiring, etc, all of which require class(es) atm.